### ISSUE TITLE
python_virtualenv seizes using recent version

### ISSUE THREAD

**Issue by gretel** [poise/poise-python#25]
`chef-12.6.0` on `ubuntu-12.04` - only thing changed is the newer version.

```
         * python_runtime_pip[2.7] action install (up to date)
        (up to date)
        (up to date)
        (up to date)
          (up to date)
       * python_virtualenv[/opt/graphite] action create
       * python_runtime_pip[/opt/graphite] action install

           ================================================================================
           Error executing action `install` on resource 'python_runtime_pip[/opt/graphite]'
           ================================================================================

           Errno::ENOENT
           -------------
           No such file or directory - /opt/graphite/bin/python

           Cookbook Trace:
           ---------------
           /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:155:in `pip_version'
       /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:67:in `block in load_current_resource'
           /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:65:in `tap'
           /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:65:in `load_current_resource'
       /tmp/kitchen/cache/cookbooks/poise/files/halite_gem/poise/helpers/notifying_block.rb:69:in `notifying_block'
           /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/python_providers/base.rb:49:in `action_install'

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/python_providers/base.rb

           116:         python_runtime_pip new_resource.name do
       117:           parent new_resource
           118:           # If the version is `true`, don't pass it at all.
           119:           version pip_version if pip_version.is_a?(String)
           120:           get_pip_url pip_url if pip_url
           121:         end
           122:       end
           123:

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/python_providers/base.rb:116:in `install_pip'

           python_runtime_pip("/opt/graphite") do
             action [:install]
             retries 0
         retry_delay 2
             default_guard_interpreter :default
         declared_type :python_runtime_pip
             cookbook_name "graphite_ng"
         parent python_virtualenv[/opt/graphite]
           end


         ================================================================================
         Error executing action `create` on resource 'python_virtualenv[/opt/graphite]'
         ================================================================================

         Errno::ENOENT
         -------------
         python_runtime_pip[/opt/graphite] (/tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/python_providers/base.rb line 116) had an error: Errno::ENOENT: No such file or directory - /opt/graphite/bin/python

         Cookbook Trace:
         ---------------
         /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:155:in `pip_version'
       /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:67:in `block in load_current_resource'
       /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:65:in `tap'
       /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/resources/python_runtime_pip.rb:65:in `load_current_resource'
         /tmp/kitchen/cache/cookbooks/poise/files/halite_gem/poise/helpers/notifying_block.rb:69:in `notifying_block'
         /tmp/kitchen/cache/cookbooks/poise-python/files/halite_gem/poise_python/python_providers/base.rb:49:in `action_install'

         Resource Declaration:
         ---------------------
         # In /tmp/kitchen/cache/cookbooks/graphite_ng/recipes/_python.rb

          35: python_virtualenv node['graphite']['base_path'] do
        36:   user node['graphite']['carbon']['user']
          37:   group node['graphite']['carbon']['group']
          38:   system_site_packages true
        39: end

         Compiled Resource:
         ------------------
         # Declared in /tmp/kitchen/cache/cookbooks/graphite_ng/recipes/_python.rb:35:in `from_file'

       python_virtualenv("/opt/graphite") do
         provider PoisePython::Resources::PythonVirtualenv::Provider
         action [:create]
           retries 0
         retry_delay 2
         default_guard_interpreter :default
         subresources [python_execute[validate-storage-schemas], python_execute[carbon-mockup], python_execute[migrate-database], python_execute[create-superuser], python_execute[validate-django], python_runtime_pip[/opt/graphite], python_package[setuptools], python_package[wheel]]
         declared_type :python_virtualenv
           cookbook_name "graphite_ng"
           recipe_name "_python"
           user "graphite"
         group "graphite"
           system_site_packages true
           parent_python python_runtime[2.7]
           path "/opt/graphite"
           pip_version true
         setuptools_version true
         wheel_version true
       end
```

**Comment by coderanger** [comment]
That usually happens because you created `/opt/graphite` via a `directory` resource earlier in the recipe code. If the folder exists, it doesn't run virtualenv/venv.

**Comment by gretel** [comment]
now that you explain it makes sense.. may i suggest adding this to the README? thanks!

**Comment by coderanger** [comment]
The plan is to change it to check the for the python binary instead because this keeps happening

**Comment by coderanger** [comment]
Released in `poise-python` 1.2.0.

**Comment by gretel** [comment]
@coderanger works like a charm, even on exisiting previously virtualenv-less systems. cheers!
